File: TumCampusApp/src/de/tum/in/tumcampusapp/models/LectureDetailsRowSet.java
package de.tum.in.tumcampusapp.models;

import java.util.List;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

/**
 * This class is dealing with the deserialization of the output of TUMOnline to
 * the method "DetailsLehrveranstaltungen"
 * 
 * @author Daniel Mayr
 * @see LectureDetailsRow
 * @see http://simple.sourceforge.net/download/stream/doc/tutorial/tutorial.php
 */
@Root(name = "rowset")
public class LectureDetailsRowSet {

	@ElementList(inline = true)
	private List<LectureDetailsRow> lehrveranstaltungenDetails;

	public List<LectureDetailsRow> getLehrveranstaltungenDetails() {
		return lehrveranstaltungenDetails;
	}

	public void setLehrveranstaltungenDetails(List<LectureDetailsRow> lehrveranstaltungenDetails) {
		this.lehrveranstaltungenDetails = lehrveranstaltungenDetails;
	}
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/lectures/model/LectureDetailsRowSet.kt
package de.tum.`in`.tumcampusapp.component.tumui.lectures.model

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml

/**
 * This class is dealing with the deserialization of the output of TUMOnline to
 * the method "DetailsLehrveranstaltungen"
 *
 * @see LectureDetailsRow
 */
@Xml(name = "rowset")
data class LectureDetailsRowSet(
        @Element var lehrveranstaltungenDetails: List<LectureDetailsRow> = mutableListOf() // TODO: Rename variable
)
